src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1138,31 +1153,24 @@ uct_dc_mlx5_iface_fc_handler(uct_rc_iface_t *rc_iface, unsigned qp_num,
                      ucs_status_string(status));
         }
     } else if (fc_hdr == UCT_RC_EP_FC_PURE_GRANT) {
-        sender = (uct_dc_fc_sender_data_t *)(hdr + 1);
-
-        it = kh_get(uct_dc_mlx5_fc_hash, &iface->tx.fc_hash, sender->ep);
-        if ((it == kh_end(&iface->tx.fc_hash)) ||
-            (kh_value(&iface->tx.fc_hash, it).seq != sender->payload.seq)) {
-            /* Just ignore, ep was removed. We either not expecting grant on
-             * this EP or this is not the same grant sequence number we are
-             * expecting. */
-            return UCS_OK;
+        sender = (uct_dc_fc_sender_data_t*)(hdr + 1);
+        ep     = (uct_dc_mlx5_ep_t*)sender->ep;
```

Comment:
`uct_dc_mlx5_fc_retrieve_ep` converts it back to `uint64`. Better do not pass ep to `uct_dc_mlx5_fc_retrieve_ep` and make all handling inside

Reply:
refactored, now it is not needed